plugins/registry_test.go

```diff
@@ -17,3 +18,12 @@ func TestValidatePlugins(t *testing.T) {
 		}
 	}
 }
+
+func TestAllPluginsHaveUniqueNames(t *testing.T) {
+	var pluginNames []string
+	for _, p := range registry {
+		pluginNames = append(pluginNames, p.Name)
+	}
+
+	assert.True(t, schema.IsStringSliceASet(pluginNames))
+}
```

Comment:
We could add a similar test that also makes sure the credential ID's used within a plugin are unique.